process/smartContract/builtInFunctions/keyValueStorage.go

```diff
@@ -32,6 +32,12 @@ func NewSaveKeyValueStorageFunc(
 	return s, nil
 }
 
+// SetNewGasConfig is called whenever gas cost is changed
+func (k *saveKeyValueStorage) SetNewGasConfig(gasCost *process.GasCost) {
+	k.funcGasCost = gasCost.BuiltInCost.SaveKeyValue
```

Comment:
We need to protect `k.funcGasCost` and `k.gasConfig` with a mutex

Reply:
no need. there is no possibility to call the from different go routines

Reply:
protected all the way